docs/en/observability/synthetics-params-secrets.asciidoc

```diff
@@ -0,0 +1,137 @@
+
+++++
+<titleabbrev>Working with parameters and secrets</titleabbrev>
+++++
+
+[discrete]
+[[synthetics-params-secrets-overview]]
+== Overview
+
+beta[] You may need to use dynamically defined values in your synthetic scripts, which may sometimes be sensitive. 
+For instance, you may want to test a production website with a special demo account whose password is only known to the team administering heartbeat. 
+Another scenario might be that you want to use a different URL when running the tests under heartbeat then running them locally using the synthetics agent. 
+Solving these problems is where params come in. Params are variables that can be used within a synthetic suite. 
+They can be defined either in the suites config file, on the synthetics agent command line, or in a heartbeat yaml config.
+
+It should also be noted that since synthetics is a full javascript environment it is also possible to use regular environment variables via
+the standard node `process.env` global object.
+
+[discrete]
+[[synthetics-basic-params]]
+=== Parameter Basics
+
+When writing a test suite, parameters can be referenced via the `params` property available within the 
+argument to a `journey`, `before`, `beforeAll`, `after`, or `afterAll` callback function.
+
+[source,js]
+----
+beforeAll(({params}) => {
+  console.log(`Visiting ${params.url}`)
+})
+
+journey("My Journey", ({ page, params }) => {
+    step('launch app', async () => {
+      await page.goto(params.url);<1>
+    });
+});
+----
+<1> Note that in a typescript program you would want to instead use `params.url as string`.
+
+If you try to run the example above you'll notice an error because we haven't specified a value for the `url` parameter.
+Parameter values can be declared by any of the following methods, listed in order of increasing precedence:
+
+* Declaring a default value for the parameter in a configuration file.
+* Passing the `--suite-params` CLI argument. 
+* Specifying the parameter in the heartbeat yaml config using the `params` option.
+
+A common pattern is to use a configuration file for default values to be configured via a configuration file, using the 
+`--suite-params` CLI argument, or `params` heartbeat option to override these defaults in the given environment. 
+
+These options are discussed in detail in the sections below.
+
+[discrete]
+[[synthetics-configs]]
+=== Using a config file to set params
+
+Use a `synthetics.config.js` or `synthetics.config.ts` file to define variables your tests always need to be defined. 
+This file should be placed in the root of your synthetics project. 
+
+[source,js]
+----
+export default (env) => {
+  let url = "http://localhost:8080";
+  if (env === "production") {
+    url = "https://elastic.github.io/synthetics-demo/"
+  }
+  return {
+    params: {
+      url,
+    },
+  };
+};
+----
+
+Note that in the above example we use the `env` variable, which corresponds to the value of the `-e` or `--environment` CLI
```

Comment:
Can we also say users can also rely on NODE_ENV for defining environments as we can deprecate this flag in future versions.